File: domain/src/main/java/com/mbmc/template/domain/usecase/RepoUseCase.java
package com.mbmc.template.domain.usecase;

import com.mbmc.template.domain.entity.Repo;
import com.mbmc.template.domain.repository.RepoRepository;
import io.reactivex.Observable;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.List;

@Singleton
public class RepoUseCase {

    private final RepoRepository repoRepository;

    @Inject
    public RepoUseCase(RepoRepository repoRepository) {
        this.repoRepository = repoRepository;
    }

    public Observable<List<Repo>> getRepos(String handle) {
        return repoRepository.getRepos(handle);
    }

}

File: domain/src/main/kotlin/com/mbmc/template/domain/usecase/RepoUseCase.kt
package com.mbmc.template.domain.usecase

import com.mbmc.template.domain.entity.RepoDomain
import com.mbmc.template.domain.repository.RepoRepository
import io.reactivex.Single

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class RepoUseCase @Inject constructor(private val repoRepository: RepoRepository) {
    fun getRepos(handle: String): Single<List<RepoDomain>> = repoRepository.getRepos(handle)
}
